plenum/server/view_change/instance_change_provider.py

```diff
@@ -89,19 +89,27 @@ def _update_votes(self, view_no: int):
     def _update_db_from_cache(self, view_no):
         if not self._instance_change_db:
             return
+        value = self._cache.get(view_no, None)
+        if not value:
+            self._instance_change_db.remove(view_no)
+            return
         serialized_value = \
-            instance_change_db_serializer.serialize(self._cache.get(view_no, None))
+            instance_change_db_serializer.serialize(value)
         if serialized_value:
             self._instance_change_db.put(str(view_no), serialized_value)
 
     def _fill_cache_by_db(self):
         if not self._instance_change_db:
             return
-        for view_no, serialized_votes in self._instance_change_db.iterator(include_value=True):
-            if not view_no.isdigit():
+        for view_no_str, serialized_votes in self._instance_change_db.iterator(include_value=True):
+            if serialized_votes is None:
+                break
+            if not view_no_str.isdigit():
                 logger.warning("InstanceChangeProvider: view_no='{}' "
-                               "must be of int type".format(view_no))
+                               "must be of int type".format(view_no_str))
                 break
+            else:
+                view_no = int(view_no_str)
```

Comment:
I think `else` is not really needed here - we already have early out if `view_no_str` is not an integer

Reply:
Yes, sorry. Fixed it.